### PR TITLE
[SYCL] Update LIT to support the latest BEs

### PR DESCRIPTION
**Pull request by vladimirlaz**
- add support for ext_intel_esimd_emulator BE. Only SYCL/ESIMD test
should be run for this BE;
 - switch to use actual BE names to avoid runtime warning;
 - updated documentation accordingly.

### CODE REVIEW

**Comment by vladimirlaz** [comment]
@dongkyunahn-intel, @kbobrovs the testing on with compiler built from intel/llvm#5058 provides the following pass rate for the emulator:
```
Testing Time: 145.49s
  Unsupported      : 18
  Passed           : 35
  Expectedly Failed:  3
  Failed           : 61
```
is that expected?

File: SYCL/CMakeLists.txt [hunk 1]
```diff
@@ -19,8 +19,12 @@ if(CHECK_SYCL_ALL)
 
     string(REPLACE "," "_" TARGET check-sycl-${TARGET_BE}-${TARGET_DEVICES})
 
+    set(TEST_DIR ".")
+    if(${TARGET_BE} MATCHES "esimd_emulator")
+      set(TEST_DIR "./ESIMD/")
+    endif()
     add_custom_target(${TARGET}
-      COMMAND python3 ${TEST_SUITE_LIT} ${TEST_SUITE_LIT_FLAGS} --param sycl_be=${TARGET_BE} --param target_devices=${TARGET_DEVICES} .
+      COMMAND python3 ${TEST_SUITE_LIT} ${TEST_SUITE_LIT_FLAGS} --param sycl_be=${TARGET_BE} --param target_devices=${TARGET_DEVICES} ${TEST_DIR}
```
Comment (bader) [inline SYCL/CMakeLists.txt]: Can we sink this logic to lit.cfg and adjust `config.test_source_root` (and probably `config.test_exec_root`)?

**Comment by vladimirlaz** [comment]
The failure is unrelated to the patch:
 - SYCL :: Plugin/level_zero_batch_barrier.cpp will be fixed after the next compiler nightly is available.

**Review by bader — DISMISSED** [review]

File: SYCL/lit.cfg.py [hunk 2]
```diff
@@ -176,37 +176,52 @@
     config.sycl_be = config.sycl_be[3:]
 config.sycl_be = config.sycl_be.lower()
 
+# Replace deprecated backend names
+deprecated_names_mapping = {'cuda'       : 'ext_oneapi_cuda',
+                            'hip'        : 'ext_oneapi_hip',
+                            'level_zero' : 'ext_oneapi_level_zero',
+                            'esimd_cpu'  : 'ext_intel_esimd_emulator'}
```
Comment (dongkyunahn-intel) [inline SYCL/lit.cfg.py:183]:
```suggestion
                            'esimd_cpu'  : 'ext_intel_esimd_emulator'}
```
```suggestion
                            'esimd_emulator'  : 'ext_intel_esimd_emulator'}
```

**Review by kbobrovs — DISMISSED** [review]

**Comment by dongkyunahn-intel** [comment]
I locally tested esimd_emulator backend with this PR and observed failures from ESIMD kernels - e.g. memory intrinsic from esimd_emulator, time-out, etc while 19 tests passed. I'm going to use my previous PR (https://github.com/intel/llvm-test-suite/pull/751) for marking such kernels accordingly - either `XFAIL` or `UNSUPPORTED`. Please merge this PR after my PR is merged.

**Comment by kbobrovs** [comment]
right, we want to start running regularly those which can run today, address the rest afterwards.

File: SYCL/lit.cfg.py (hunk 2 above)
Comment (vladimirlaz) [reply SYCL/lit.cfg.py:183]: here old name of the ESIMD emulator was used https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/backend_types.hpp#L31

Comment (kbobrovs) [inline SYCL/lit.cfg.py:183]: I wonder BTW, why would we keep the deprecated esimd_cpu name rather than simply drop it?

**Review by kbobrovs — APPROVED** [review]
LGTM, with one comment

**Review by bader — APPROVED** [review]

File: SYCL/lit.cfg.py (hunk 2 above)
Comment (vladimirlaz) [reply SYCL/lit.cfg.py:183]: we support all deprecated BE until they are finally removed.

**Pull request merged**